### PR TITLE
Disable pool&conv_transpose&quantize caching

### PR DESCRIPTION
**Pull request by jczaja**
### PR types
Bug fixes

### PR changes
OPs

### Describe
PaddlePaddle is caching oneDNN objects and this PR is to disable this for pool2d and conv_transpose. Reason is that oneDNN is caching some of its objects itself

### CODE REVIEW

**Comment by jczaja** [comment]
@wozna Please take a look at changes in quantize.

**Comment by jczaja** [comment]
@tsocha Please review

**Review by tsocha — DISMISSED** [review]

File: paddle/fluid/operators/mkldnn/conv_transpose_mkldnn_op.cc [hunk 1]
```diff
@@ -217,86 +210,140 @@ class ConvTransposeMKLDNNHandlerT
...
+      int c = iohw_weights_tz[0];
+      int h = iohw_weights_tz[2];
+      int w = iohw_weights_tz[3];
+      std::shared_ptr<K> reordered_filter_data(new K[o * c * h * w](),
+                                               std::default_delete<K[]>());
+      for (int i = 0; i < c; ++i) {
+        for (int j = 0; j < o; ++j) {
+          int in_offset = j * h * w + i * o * h * w;
+          int out_offset = j * c * h * w + i * h * w;
+          std::memcpy(&(reordered_filter_data.get())[out_offset],
```
Comment (jakpiase) [inline paddle/fluid/operators/mkldnn/conv_transpose_mkldnn_op.cc:246]: I know that might not be your code, but shouldn't that reorder be done using oneDNN?

Comment (jczaja) [reply paddle/fluid/operators/mkldnn/conv_transpose_mkldnn_op.cc:246]: :) @Silv3S is working on that.

**Review by lidanqing-vv — DISMISSED** [review]
LGTM

**Comment by jczaja** [comment]
@chenwhql Could you please approve so that PR-CI-APPROVAL pass? PADDLE_ENFORCE is correct but due only part of it is visible in diff, corresponding checker cannot see full statement and report error.

**Comment by lidanqing-vv** [comment]
@baoachun @Superjomn This PR is part of the improving cache mechanism. CI reported PADDLE_ENFORCE error but actually it is correct but due only part of it is visible in diff, corresponding checker cannot see full statement and report error.
Could you please help approve it? Thanks.

File: paddle/fluid/operators/mkldnn/conv_transpose_mkldnn_op.cc [hunk 2]
```diff
@@ -40,151 +40,144 @@ inline mkldnn::memory::dims GetWeightsTz(const Tensor* filter,
...
           platform::errors::InvalidArgument(
-              "The filter tensor's laytout should be %d, but got %d.",
-              DataLayout::kMKLDNN, filter->layout()));
-      PADDLE_ENFORCE_NE(filter->format(), MKLDNNMemoryFormat::undef,
+              "The bias tensor's laytout should be %d, but got %d.",
+              DataLayout::kMKLDNN, bias->layout()));
+      PADDLE_ENFORCE_NE(bias->format(), MKLDNNMemoryFormat::undef,
                         platform::errors::InvalidArgument(
-                            "Got wrong formats for Filter tensor."));
+                            "Got wrong format for Bias tensor."));
```
Comment (lidanqing-vv) [inline paddle/fluid/operators/mkldnn/conv_transpose_mkldnn_op.cc:96]: @baoachun

**Review by wozna — DISMISSED** [review]
LGTM

**Review by jakpiase — DISMISSED** [review]
LGTM

File: paddle/fluid/operators/mkldnn/conv_transpose_mkldnn_op.cc (hunk 2 above)
Comment (lidanqing-vv) [reply paddle/fluid/operators/mkldnn/conv_transpose_mkldnn_op.cc:96]: image
Hi, @baoachun 
这里，错误信息长度符合要求，要求至少20个字符长度，这里错误信息有30个字符左右长度。这里的问题是PADDLE_ENFORCE 机制不认未更改的行，这里 95行 `platform::errors::InvalidArgument(`已经写了，只是这个PR没有更改，所以就没检测到，图片里InvalidArgument代码就没有。但是其实代码中是有的，是PADLLE_ENFORCE机制没检测到。

Comment (lidanqing-vv) [reply paddle/fluid/operators/mkldnn/conv_transpose_mkldnn_op.cc:96]: 这个PADDLE_ENFORCE机制以后会考虑优化吗，以后可能也经常有类似问题。

Comment (chenwhql) [reply paddle/fluid/operators/mkldnn/conv_transpose_mkldnn_op.cc:96]: 这里是字符串正则匹配检测的，先grep找出修改了的行，然后正则匹配做合规检查，如果全文件正则检查，可以避免这个问题，但是可能会检出和PR不相关的一些问题，不太友好。。。目前这一点没有想到好的办法处理，先找我approve吧

**Review by chenwhql — DISMISSED** [review]
LGTM for PADDLE_ENFORCE

File: paddle/fluid/operators/mkldnn/conv_transpose_mkldnn_op.cc (hunk 2 above)
Comment (lidanqing-vv) [reply paddle/fluid/operators/mkldnn/conv_transpose_mkldnn_op.cc:96]: 好的，非常感谢！

**Comment by lidanqing-vv** [comment]
@jczaja This PR is approved by Baidu. But could you please watch PR-CI-iScan-C to pass ? Cause only this one keep failing.

**Comment by jczaja** [comment]
@baoachun Please help with failing CI (it did pass few reruns ago). Error is: "git code download failed ". I have try couple of reruns but without success.

**Comment by baoachun** [comment]
> @baoachun Please help with failing CI (it did pass few reruns ago). Error is: "git code download failed ". I have try couple of reruns but without success.

My colleague is solving the problem, please wait.

**Comment by baoachun** [comment]
Hi @jczaja, my colleague said that you may need to pull the latest code of the develop branch.

**Comment by jczaja** [comment]
@baoachun  I have rebased this PR but still PR-CI-iScan-C  is failing (I have done one rerurn). Please help

**Review by chenwhql — APPROVED** [review]

**Pull request merged**